public/components/ui_action_context_menu.tsx

```diff
@@ -75,13 +99,15 @@ export const ActionContextMenu = (props: Props) => {
     [actionContext.datasetId, actionContext.datasetType, actionContext.dataSourceId]
   );
 
-  // The action button should be not displayed when there is no action and result summary disabled.
-  if (actionsRef.current.length === 0 && !resultSummaryEnabled) {
+  // The action button should be not displayed when there is no action and result summary disabled or there is no data2Summary agent
+  if ((!resultSummaryEnabled || !isSummaryAgentAvailable) && actionsRef.current.length === 0) {
     return null;
   }
 
-  // The action button should be disabled when context menu has no item and result summary disabled.
-  const actionDisabled = (panels.value?.[0]?.items ?? []).length === 0 && !resultSummaryEnabled;
```

Comment:
Can be replace with `(panels.value?.[0]?.items ?? []).length === 0 && ! shouldShowSummarizationAction`.